pkg/virtual/framework/dynamic/apiserver/handler_test.go

```diff
@@ -251,11 +336,29 @@ func TestRouting(t *testing.T) {
 			ExpectDelegateCalled: false,
 			ExpectStatus:         200,
 			ExpectResponse: func(t *testing.T, r *http.Response, b []byte) {
-				if r.Header.Get("Content-Type") == "application/json" && r.StatusCode == 200 {
-					require.Equal(t,
-						"{\"kind\":\"APIResourceList\",\"apiVersion\":\"v1\",\"groupVersion\":\"custom/v1\",\"resources\":[{\"name\":\"customresources\",\"singularName\":\"customresource\",\"namespaced\":true,\"kind\":\"CustomResource\",\"verbs\":[\"get\",\"list\",\"patch\",\"create\",\"update\",\"watch\"],\"storageVersionHash\":\"ixY6U/JU9OM=\"}]}\n",
-						string(b))
+				if r.Header.Get("Content-Type") != "application/json" || r.StatusCode != 200 {
+					// why?
+					return
 				}
+				var list metav1.APIResourceList
+				require.NoError(t, json.Unmarshal(b, &list))
+				require.Empty(t, cmp.Diff(list, metav1.APIResourceList{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "APIResourceList",
+						APIVersion: "v1",
+					},
+					GroupVersion: "custom/v1",
+					APIResources: []metav1.APIResource{
+						{
+							Name:               "customresources",
+							SingularName:       "customresource",
+							Namespaced:         true,
+							Kind:               "CustomResource",
+							Verbs:              []string{"get", "list"},
```

Comment:
This just parrots whatever data we put into the mocks, perhaps we should use resource names etc that don't have any meaning, so as to not confuse people - certainly I was surprised to learn this was not reading some real store of CRD information.